library/testing/PseudoUnitTest.h

```diff
@@ -0,0 +1,150 @@
+#ifndef Test_h
+#define Test_h
+
+#include <functional>
+#include <iostream>
+#include <sstream>
+
+/** Helper to perform multiple checks within the same `ctest` test.
+ * Checks are performed using the various overloads of `operator()`
+ * and their results are logged and tracked so that the `result()` method
+ * will return `EXIT_FAILURE` if any of them failed (and `EXIT_SUCCESS` otherwise).
+ */
+class PseudoUnitTest
+{
+public:
+  /** test a boolean condition */
+  void operator()(const std::string& label, bool condition)
```

Comment:
I suppose the argument is always a string view rvalue, so I think this make more sense (here and everywhere):
```suggestion
  void operator()(std::string_view&& label, bool condition)
```

Reply:
leaving this one out because it's c++17 only